rollup/da_syncer/l1rpc_fetcher.go

```diff
@@ -0,0 +1,262 @@
+package da_syncer
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"math/big"
+	"reflect"
+
+	"github.com/scroll-tech/go-ethereum/accounts/abi"
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/core/rawdb"
+	"github.com/scroll-tech/go-ethereum/core/types"
+	"github.com/scroll-tech/go-ethereum/ethdb"
+	"github.com/scroll-tech/go-ethereum/log"
+	"github.com/scroll-tech/go-ethereum/params"
+	"github.com/scroll-tech/go-ethereum/rollup/sync_service"
+)
+
+type L1RPCFetcher struct {
+	fetchBlockRange               uint64
+	client                        *L1Client
+	db                            ethdb.Database
+	ctx                           context.Context
+	latestProcessedBlock          uint64
+	scrollChainABI                *abi.ABI
+	l1CommitBatchEventSignature   common.Hash
+	l1RevertBatchEventSignature   common.Hash
+	l1FinalizeBatchEventSignature common.Hash
+}
+
+func newL1RpcDaFetcher(ctx context.Context, genesisConfig *params.ChainConfig, l1Client sync_service.EthClient, db ethdb.Database, l1DeploymentBlock, fetchBlockRange uint64) (DaFetcher, error) {
+	// terminate if the caller does not provide an L1 client (e.g. in tests)
+	if l1Client == nil || (reflect.ValueOf(l1Client).Kind() == reflect.Ptr && reflect.ValueOf(l1Client).IsNil()) {
```

Comment:
the use of reflect is generally not necessary for some checks.

Reply:
`sync_service.EthClient` is interface, so `l1Client` may be equal nil, or may be of type nil

Reply:
sure, but that's bad programming if you nil it. 
In general, just checking for nil is sufficient for all those cases, but it's also ok if you want to keep it.

Reply:
also, don't do that in perf critical code (it isn't here, so it's fine) as reflect really is expensive in general.